.circleci/docker/common/install_base.sh

```diff
@@ -2,52 +2,115 @@
-# for this case we will set UBUNTU_VERSION to `18.04-rc` so that the Dockerfile could
-# find the correct image. As a result, here we have to check for
-#   "$UBUNTU_VERSION" == "18.04"*
-# instead of
-#   "$UBUNTU_VERSION" == "18.04"
-if [[ "$UBUNTU_VERSION" == "18.04"* ]]; then
-  cmake3="cmake=3.10*"
+install_ubuntu() {
+  # NVIDIA dockers for RC releases use tag names like `11.0-cudnn8-devel-ubuntu18.04-rc`,
+  # for this case we will set UBUNTU_VERSION to `18.04-rc` so that the Dockerfile could
+  # find the correct image. As a result, here we have to check for
+  #   "$UBUNTU_VERSION" == "18.04"*
+  # instead of
+  #   "$UBUNTU_VERSION" == "18.04"
+  if [[ "$UBUNTU_VERSION" == "18.04"* ]]; then
+    cmake3="cmake=3.10*"
+  else
+    cmake3="cmake=3.5*"
+  fi
+
+  # Install common dependencies
+  apt-get update
+  # TODO: Some of these may not be necessary
+  # TODO: libiomp also gets installed by conda, aka there's a conflict
+  ccache_deps="asciidoc docbook-xml docbook-xsl xsltproc"
+  numpy_deps="gfortran"
+  apt-get install -y --no-install-recommends \
+    $ccache_deps \
+    $numpy_deps \
+    ${cmake3} \
+    apt-transport-https \
+    autoconf \
+    automake \
+    build-essential \
+    ca-certificates \
+    curl \
+    git \
+    libatlas-base-dev \
+    libc6-dbg \
+    libiomp-dev \
+    libyaml-dev \
+    libz-dev \
+    libjpeg-dev \
+    libasound2-dev \
+    libsndfile-dev \
+    python \
+    python-dev \
+    python-setuptools \
+    python-wheel \
+    software-properties-common \
+    sudo \
+    wget \
+    vim
+
+  # TODO: THIS IS A HACK!!!
+  # distributed nccl(2) tests are a bit busted, see https://github.com/pytorch/pytorch/issues/5877
+  if dpkg -s libnccl-dev; then
+    apt-get remove -y libnccl-dev libnccl2 --allow-change-held-packages
+  fi
+
+  # Cleanup package manager
+  apt-get autoclean && apt-get clean
+  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+}
+
+install_centos() {
+  # Need EPEL for many packages we depend on.
+  # See http://fedoraproject.org/wiki/EPEL
+  yum --enablerepo=extras install -y epel-release
+
+  ccache_deps="asciidoc docbook-dtds docbook-style-xsl libxslt"
+  numpy_deps="gcc-gfortran"
+  # Note: protobuf-c-{compiler,devel} on CentOS are too old to be used
+  # for Caffe2. That said, we still install them to make sure the build
+  # system opts to build/use protoc and libprotobuf from third-party.
+  yum install -y \
+    $ccache_deps \
+    $numpy_deps \
+    autoconf \
+    automake \
+    bzip2 \
+    cmake \
+    cmake3 \
+    curl \
+    gcc \
+    gcc-c++ \
+    gflags-devel \
+    git \
+    glibc-devel \
+    glibc-headers \
+    glog-devel \
+    hiredis-devel \
+    libstdc++-devel \
+    make \
+    opencv-devel \
+    sudo \
+    wget \
+    vim
+
+  # Cleanup
+  yum clean all
+  rm -rf /var/cache/yum
+  rm -rf /var/lib/yum/yumdb
+  rm -rf /var/lib/yum/history
+}
+
+# Install base packages depending on the base OS
+if [ -f /etc/lsb-release ]; then
```

Comment:
this is a brittle way to do this as well since there is a chance (albeit small) that centos can also contain an `lsb-release` file.

I'd rather base this is off of the `$ID` variable in `/etc/os-release` since that's more consistent, that'll also allow you to use a `case statement here as well instead of a bunch of `if` statements.

Reply:
Resolved by 68d0c0e98d967c910b64aacfe9b46042cad3e741.  Please review.